sa/model.go

```diff
@@ -1055,26 +1055,29 @@ func deleteOrderFQDNSet(
 }
 
 func addIssuedNames(ctx context.Context, queryer db.Execer, cert *x509.Certificate, isRenewal bool) error {
-	if len(cert.DNSNames) == 0 {
-		return berrors.InternalServerError("certificate has no DNSNames")
-	}
-
-	multiInserter, err := db.NewMultiInserter("issuedNames", []string{"reversedName", "serial", "notBefore", "renewal"})
-	if err != nil {
-		return err
-	}
-	for _, name := range cert.DNSNames {
-		err = multiInserter.Add([]interface{}{
-			ReverseName(name),
-			core.SerialToString(cert.SerialNumber),
-			cert.NotBefore.Truncate(24 * time.Hour),
-			isRenewal,
-		})
+	// TODO(#7311): Determine & explicitly document whether to place IP address
+	// identifiers in issuedNames. We currently skip them.
+	if len(cert.DNSNames) > 0 {
```

Comment:
In combination with my comment below, we don't need these conditionals. Just loop over the DNSNames, then loop over the IPAddresses, and if either of those is empty then that loop will run zero times.